pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -95,6 +95,8 @@ pub fn spawn_connection_manager_task(
     );
 }
 
+const MAX_BROKER_IDLE_WAIT: Duration = Duration::from_secs(90);
```

Comment:
https://neon-github-public-dev.s3.amazonaws.com/reports/pr-3147/debug/3733221379/index.html#suites/158be07438eb5188d40b466b6acfaeb3/85f48b6cc4d58c58/

There are surprisingly many tests that manage to get idle during tests. 90 seconds produces two tests in debug only, but when I use 60 seconds, I get 7 tests in Debug runs:

https://neon-github-public-dev.s3.amazonaws.com/reports/pr-3147/debug/3731383425/index.html#suites/3de20ced56df482626cff043a3f2c817

What would be the realistic value, if the PR's approach is correct?

Generally, I'm not sure about the `WARN` message level too: IIRC, safekeepers should have the heuristics to stop sending WAL events if no new data arrives. 
So, maybe turn this to `INFO` and go on?

Reply:
It's actually ok to have no new events (idle) from the broker when compute is down. That means that current approach can spam with `Got no broker or walreceiver events ...` in the logs, because we always have thousands of timelines with no running compute.

PR's approach does not feel right. IMO would be better to put more walreceiver context into `wait_lsn` instead, e.g. we can print latest broker event and full walreceiver status on WAL timeout error.

Reply:
> better to put more walreceiver context into `wait_lsn` instead

Agree that we should do it, planned to do in a separate issue here: https://github.com/neondatabase/neon/issues/2106

What I wanted to have with this log is to catch slow walreceiver reconnects in our tests.
IMO, there's definitely something abnormal: we don't shut down *all* SK nodes for 60 seconds in 7 tests, yet we cannot get broker messages there for some reason.

I'll think of it more, will remove the logging and will try to add some metric instead (and, maybe, debug logging).

Reply:
I tried to run one failed test locally, and it seems we have tests with a root timeline created by `env = neon_env_builder.init_start()`, which is never used after that. We do that in all failed tests, and that timeline has a spawned walreceiver with no activity (no one starts a compute for it). Without a compute safekeepers don't even know about that timeline and don't push anything to the broker.

It looks like these tests fail because they run longer than MAX_BROKER_IDLE_WAIT.

Reply:
Yes, just realized the same when discussed the issue recently, thank you for confirming.
Thinking of it more, I agree now that we should remove the related commit entirely and work on https://github.com/neondatabase/neon/issues/2106 (https://github.com/neondatabase/neon/issues/3115 later, maybe metrics too?) to get the required info, including the "time spent without etcd events" one.

Reply:
Have pushed another commit, that removes this logging entirely, added a few `debug!` logs in case that place would need it later.